src/datasources/email-template/pushwoosh-template.service.ts

```diff
@@ -1,24 +1,75 @@
-import { Injectable } from '@nestjs/common';
+import { Inject, Injectable } from '@nestjs/common';
 import { getAddress } from 'viem';
 import { IEmailTemplate } from '@/domain/interfaces/email-template.interface';
+import { IChainsRepository } from '@/domain/chains/chains.repository.interface';
 
 @Injectable()
 export class PushWooshTemplate implements IEmailTemplate {
-  addressToHtml(address: string): string {
+  constructor(
+    @Inject(IChainsRepository)
```

Comment:
I think we don't have an hard rule for this, but I'd try to not inject a Repository at this level (datasource layer). 

What do you think about retrieving the `Chain` object and passing it to the methods in `IEmailTemplate` (i.e.: passing `chain: Chain` instead of `chainId:  string`)? That would also have other performance-wise beneficial side-effects, because `AlertsRepository._notifySafeSetup` for example calls the three functions of `IEmailTemplate`, and with the current implementation it would translate into three calls to the repository, which could be just one by retrieving the `Chain` beforehand.

Reply:
I've adjusted the code to pass the `Chain` as an argument in f08a727f880da61260c379da75737d62ecca97c0. As well as the performance improvement, it simplified some of the code as well.